pyteal/compiler/compiler_test.py

```diff
@@ -3813,504 +2441,29 @@ def log_creation(*, output: pt.abi.String) -> pt.Expr:
-frame_bury 2
-frame_dig 1
-frame_dig 2
-callsub div_3
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// mod_caster
-modcaster_13:
-proto 0 0
-int 0
-dupn 2
-txna ApplicationArgs 1
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-frame_dig 1
-frame_dig 2
-callsub mod_4
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// all_laid_to_args_caster
-alllaidtoargscaster_14:
-proto 0 0
-int 0
-dupn 16
-byte ""
-txna ApplicationArgs 1
-btoi
-frame_bury 1
-txna ApplicationArgs 2
-btoi
-frame_bury 2
-txna ApplicationArgs 3
-btoi
-frame_bury 3
-txna ApplicationArgs 4
-btoi
-frame_bury 4
-txna ApplicationArgs 5
-btoi
-frame_bury 5
-txna ApplicationArgs 6
-btoi
-frame_bury 6
-txna ApplicationArgs 7
-btoi
-frame_bury 7
-txna ApplicationArgs 8
-btoi
-frame_bury 8
-txna ApplicationArgs 9
-btoi
-frame_bury 9
-txna ApplicationArgs 10
-btoi
-frame_bury 10
-txna ApplicationArgs 11
-btoi
-frame_bury 11
-txna ApplicationArgs 12
-btoi
-frame_bury 12
-txna ApplicationArgs 13
-btoi
-frame_bury 13
-txna ApplicationArgs 14
-btoi
-frame_bury 14
-txna ApplicationArgs 15
-frame_bury 17
-frame_dig 17
-int 0
-extract_uint64
-frame_bury 15
-frame_dig 17
-int 8
-extract_uint64
-frame_bury 16
-frame_dig 1
-frame_dig 2
-frame_dig 3
-frame_dig 4
-frame_dig 5
-frame_dig 6
-frame_dig 7
-frame_dig 8
-frame_dig 9
-frame_dig 10
-frame_dig 11
-frame_dig 12
-frame_dig 13
-frame_dig 14
-frame_dig 15
-frame_dig 16
-callsub alllaidtoargs_5
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// empty_return_subroutine_caster
-emptyreturnsubroutinecaster_15:
-proto 0 0
-callsub emptyreturnsubroutine_6
-retsub
-
-// log_1_caster
-log1caster_16:
-proto 0 0
-int 0
-callsub log1_7
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-itob
-concat
-log
-retsub
-
-// log_creation_caster
-logcreationcaster_17:
-proto 0 0
-byte ""
-callsub logcreation_8
-frame_bury 0
-byte 0x151f7c75
-frame_dig 0
-concat
-log
-retsub""".strip()
+    with open(FIXTURES / "yaccFP_approval_v8.teal") as f:
+        expected_ap_without_oc = f.read()
+    with open(FIXTURES / "yaccFP_clear_v8.teal") as f:
+        expected_csp_without_oc = f.read()
     assert actual_ap_without_oc_compiled == expected_ap_without_oc
+    assert actual_csp_without_oc_compiled == expected_csp_without_oc
+    append_router_info(
+        (
+            "yacc",
+            8,
+            _router_without_oc,
+        ),
+        (
+            actual_ap_without_oc_compiled,
+            actual_csp_without_oc_compiled,
+        ),
+    )
+    assert actual_csp_without_oc_compiled == expected_csp_without_oc
 
-    expected_csp_without_oc = """#pragma version 8
-int 1
-return""".strip()
+    return routers, sources
 
-    assert actual_csp_without_oc_compiled == expected_csp_without_oc
+
+def test_router_app():
+    # TODO: this test is redundant as router_app_tester is imported and run by
+    # tests/integration/abi_router_test.py's setup
+    router_app_tester()
```

Comment:
If we follow this advice and remove `test_router_app()` then most of the machinery here becomes quite strange to keep around.

Reply:
@ahangsu - I propose keeping this as is for now. #650 also makes use of this test, and maybe we can determine it's final placement during that PR discussion.

Additionally, it would be nice to test the case of a non-trivial clear program as well.

Reply:
ok, I will look into this closer, as I finished my pass on graviton.